packages/nimble-components/src/rich-text-viewer/tests/rich-text-viewer.spec.ts

```diff
@@ -193,6 +193,44 @@ describe('RichTextViewer', () => {
             ]);
         });
 
+        it('nested numbered lists markdown string to "ol" and "li" HTML tags', async () => {
+            element.markdown = '1. Option 1\n\n    1. Option 2';
+
+            await connect();
+
+            expect(pageObject.getRenderedMarkdownTagNames()).toEqual([
+                'OL',
+                'LI',
+                'P',
+                'OL',
+                'LI',
+                'P'
```

Comment:
> (when I tried creating a sequential list by adding newlines in between two lists, the component combined them into one list)

As mentioned above, it is not possible to have the same sequential lists in the markdown format, it will add up to the existing list. The only way to have this same sequential order is to have any node between the lists.

```html
<ol>
    <li>
    </li>
</ol>
<p></p>
<ol>
    <li>
    </li>
</ol>
```

For the above case, we have written the test cases [here](https://github.com/ni/nimble/blob/aa4045c729938bb2fd8743ee830d3cb96f78263f/packages/nimble-components/src/rich-text-viewer/tests/rich-text-viewer.spec.ts#L270C9-L289C12).

However, if we have sequential different lists like the one below, we will have the same outputs for both sequential and nested lists which is a gap in the testing approach.
```html
<ol>
    <li>
    </li>
</ol>
<ul>
    <li>
    </li>
</ul>
```

Therefore, I have written a helper method in the page object `getRenderedMarkdownTagNamesWithClosingTags()` which will return closing tags in a right place of an array. I have used this method exclusively for the above case we discussed on nested and sequential lists and for others I used the `getRenderedMarkdownTagNames()` method without closing tags.